Packs/ApiModules/Scripts/MicrosoftGraphMailApiModule/MicrosoftGraphMailApiModule.py

```diff
@@ -0,0 +1,1809 @@
+                'attachmentItem': {
+                    'attachmentType': 'file',
+                    'name': attachment_name,
+                    'size': attachment_size,
+                    'isInline': is_inline
+                }
+            }
+        )
+
+    def add_attachment_with_upload_session(self, email: str, draft_id: str, attachment_data: bytes,
+                                           attachment_name: str, is_inline: bool = False):
+        """
+        Add an attachment using an upload session by dividing the file bytes into chunks and sent each chunk each time.
+        more info here - https://docs.microsoft.com/en-us/graph/outlook-large-attachments?tabs=http
+        Args:
+            email (str): email to create the upload session.
+            draft_id (str): draft ID to add the attachments to.
+            attachment_data (bytes) : attachment data in bytes.
+            attachment_name (str): attachment name.
+            is_inline (bool): is the attachment inline, True if yes, False if not.
+        """
+
+        attachment_size = len(attachment_data)
+        try:
+            upload_session = self.get_upload_session(
+                email=email,
+                draft_id=draft_id,
+                attachment_name=attachment_name,
+                attachment_size=attachment_size,
+                is_inline=is_inline
+            )
+            upload_url = upload_session.get('uploadUrl')
+            if not upload_url:
+                raise Exception(f'Cannot get upload URL for attachment {attachment_name}')
+
+            start_chunk_index = 0
+            end_chunk_index = self.MAX_ATTACHMENT_SIZE
+
+            chunk_data = attachment_data[start_chunk_index: end_chunk_index]
+
+            response = self.upload_attachment(
+                upload_url=upload_url,
+                start_chunk_idx=start_chunk_index,
+                end_chunk_idx=end_chunk_index,
+                chunk_data=chunk_data,
+                attachment_size=attachment_size
+            )
+            while response.status_code != 201:  # the api returns 201 when the file is created at the draft message
+                start_chunk_index = end_chunk_index
+                next_chunk = end_chunk_index + self.MAX_ATTACHMENT_SIZE
+                end_chunk_index = next_chunk if next_chunk < attachment_size else attachment_size
+
+                chunk_data = attachment_data[start_chunk_index: end_chunk_index]
+
+                response = self.upload_attachment(
+                    upload_url=upload_url,
+                    start_chunk_idx=start_chunk_index,
+                    end_chunk_idx=end_chunk_index,
+                    chunk_data=chunk_data,
+                    attachment_size=attachment_size
+                )
+
+                if response.status_code not in (201, 200):
+                    raise Exception(f'{response.json()}')
+
+        except Exception as e:
+            demisto.error(f'{e}')
+            raise e
```

Comment:
There's an error handling in the main. Is there any reason for this additional error handling?